packages/core/test/features/st-var.spec.ts

```diff
@@ -1540,11 +1559,29 @@ describe(`features/st-var`, () => {
                     input: {
                         type: 'st-map',
                         value: {
-                            a: 'red',
+                            a: {
```

Comment:
This is where we actually want a string. 
We can check if the type is "primitive" or check specific for "st-string" and use it's `flatValue`.
When we add types to values (like color in this case) we would print the full type.

One note on `st-string` here. I'm not sure if a value with quotes is `st-string` or just any value. because we might want to change it.

Reply:
I think st-string is what you would always want here,  if we had a type for color then the type would change to color.  

In what scenario would you prefer to have a flat value instead? (which you have on the value field already)

Reply:
Not sure....it means that all of these are st-string

```
:vars {
    a: "'red'";
    b: "red";
    c: red;
}
```

Reply:
I'm not sure we need to have a difference in types for these cases. Quotations are used to wrap strings that would otherwise break the syntax. but in the end the result is the same. In any case this PR didn't change the handling of strings, just maps.